Fix `@Disabled` tests

Found when implementing https://github.com/jonapoul/modular/issues/242 (in https://github.com/jonapoul/modular/pull/248)

Fix script compilation errors from 3 plugin's DSL functions on the same classpath.

Search for the `@Disabled` tests and re-check.

```

Unexpected build execution failure in /tmp/junit-7184025186912359810 with arguments [modularGenerate, --configuration-cache, -Pandroid.useAndroidX=true]

Output:
Calculating task graph as no cached configuration is available for tasks: modularGenerate

> Configure project :
e: file:///tmp/junit-7184025186912359810/build.gradle.kts:10:5: Unresolved reference. None of the following candidates is applicable because of a receiver type mismatch:
fun D2NamedLinkTypeContainer.api(style: LinkStyle? = ..., color: String? = ..., displayName: String = ...): NamedDomainObjectProvider<LinkTypeSpec>
e: file:///tmp/junit-7184025186912359810/build.gradle.kts:11:5: Unresolved reference. None of the following candidates is applicable because of a receiver type mismatch:
fun D2NamedLinkTypeContainer.implementation(style: LinkStyle? = ..., color: String? = ..., displayName: String = ...): NamedDomainObjectProvider<LinkTypeSpec>

FAILURE: Build failed with an exception.

* Where:
Build file '/tmp/junit-7184025186912359810/build.gradle.kts' line: 10

* What went wrong:
Script compilation errors:

  Line 10:     api(color = "green")
               ^ Unresolved reference. None of the following candidates is applicable because of a receiver type mismatch:
                   fun D2NamedLinkTypeContainer.api(style: LinkStyle? = ..., color: String? = ..., displayName: String = ...): NamedDomainObjectProvider<LinkTypeSpec>

  Line 11:     implementation(color = "#5555FF")
               ^ Unresolved reference. None of the following candidates is applicable because of a receiver type mismatch:
                   fun D2NamedLinkTypeContainer.implementation(style: LinkStyle? = ..., color: String? = ..., displayName: String = ...): NamedDomainObjectProvider<LinkTypeSpec>

2 errors
```

--- a/modular-test/src/test/kotlin/modular/core/WriteReadmeTest.kt
+++ b/modular-test/src/test/kotlin/modular/core/WriteReadmeTest.kt
@@ -15,7 +15,6 @@ import modular.test.scenarios.MermaidBasic
 import modular.test.scenarios.MermaidWithLinkTypes
 import modular.test.taskHadResult
 import org.gradle.testkit.runner.TaskOutcome.SUCCESS
-import org.junit.jupiter.api.Disabled
 import kotlin.test.Test
 
 internal class WriteReadmeTest : ScenarioTest() {
@@ -46,7 +45,6 @@ internal class WriteReadmeTest : ScenarioTest() {
     )
   }
 
-  @Disabled("https://github.com/jonapoul/modular/issues/247")
   @Test
   fun `Write mermaid readme with link types`() = runScenario(MermaidWithLinkTypes) {
     // when
@@ -61,15 +59,11 @@ internal class WriteReadmeTest : ScenarioTest() {
         # a
 
         <!--region chart-->
-
         ```mermaid
         graph TD
           _a[":a"]
           _b[":b"]
           _c[":c"]
-          style _a color:black,font-weight:bold,stroke:black,stroke-width:2px
-          style _b color:black
-          style _c color:black
           _a --> _b
           linkStyle 0 stroke:green
           _a --> _c
@@ -80,13 +74,12 @@ internal class WriteReadmeTest : ScenarioTest() {
         |:--:|:--:|
         | api | Green |
         | implementation | #5555FF |
-        | compileOnly | Yellow Dotted |
+        | compileOnly | Yellow Dashed |
         <!--endregion-->
       """.trimIndent(),
     )
   }
 
-  @Disabled("https://github.com/jonapoul/modular/issues/247")
   @Test
   fun `Inject mermaid into existing readme`() = runScenario(MermaidWithLinkTypes) {
     // given
@@ -117,15 +110,11 @@ internal class WriteReadmeTest : ScenarioTest() {
       Some prefix
 
       <!--region chart-->
-
       ```mermaid
       graph TD
         _a[":a"]
         _b[":b"]
         _c[":c"]
-        style _a color:black,font-weight:bold,stroke:black,stroke-width:2px
-        style _b color:black
-        style _c color:black
         _a --> _b
         linkStyle 0 stroke:green
         _a --> _c
@@ -136,7 +125,7 @@ internal class WriteReadmeTest : ScenarioTest() {
       |:--:|:--:|
       | api | Green |
       | implementation | #5555FF |
-      | compileOnly | Yellow Dotted |
+      | compileOnly | Yellow Dashed |
       <!--endregion-->
 
       Some suffix

--- a/modular-test/src/test/kotlin/modular/graphviz/tasks/WriteGraphvizChartTest.kt
+++ b/modular-test/src/test/kotlin/modular/graphviz/tasks/WriteGraphvizChartTest.kt
@@ -16,7 +16,6 @@ import modular.test.scenarios.GraphVizChartWithReplacements
 import modular.test.scenarios.GraphvizBasic
 import modular.test.scenarios.GraphvizNestedModules
 import modular.test.scenarios.GraphvizNestedModulesNoModuleTypes
-import org.junit.jupiter.api.Disabled
 import kotlin.test.Test
 
 internal class WriteGraphvizChartTest : ScenarioTest() {
@@ -165,7 +164,6 @@ internal class WriteGraphvizChartTest : ScenarioTest() {
   }
 
   @Test
-  @Disabled("https://github.com/jonapoul/modular/issues/247")
   fun `Handle custom link types`() = runScenario(GraphVizChartWithCustomLinkTypes) {
     // when
     runTask("writeGraphvizChart").build()
@@ -178,10 +176,10 @@ internal class WriteGraphvizChartTest : ScenarioTest() {
     assertThat(dotFile.readText()).equalsDiffed(
       """
         digraph {
-          ":a" [fillcolor="mediumorchid",penwidth="3"]
-          ":b" [fillcolor="mediumorchid",shape="none"]
-          ":c" [fillcolor="orange",shape="none"]
-          ":d" [fillcolor="orange",shape="none"]
+          ":a" [fillcolor="mediumorchid"]
+          ":b" [fillcolor="mediumorchid"]
+          ":c" [fillcolor="orange"]
+          ":d" [fillcolor="orange"]
           ":a" -> ":b" [style="bold"]
           ":a" -> ":c" [color="blue"]
           ":a" -> ":d" [style="dotted",color="#FF55FF"]

--- a/modular-test/src/test/kotlin/modular/graphviz/tasks/WriteGraphvizLegendTest.kt
+++ b/modular-test/src/test/kotlin/modular/graphviz/tasks/WriteGraphvizLegendTest.kt
@@ -12,7 +12,6 @@ import modular.test.contentEquals
 import modular.test.runTask
 import modular.test.scenarios.GraphVizWithLinkTypes
 import modular.test.scenarios.GraphvizBasic
-import org.junit.jupiter.api.Disabled
 import kotlin.test.Test
 
 internal class WriteGraphvizLegendTest : ScenarioTest() {
@@ -44,7 +43,6 @@ internal class WriteGraphvizLegendTest : ScenarioTest() {
   }
 
   @Test
-  @Disabled("https://github.com/jonapoul/modular/issues/247")
   fun `Show modules and links next to each other`() = runScenario(GraphVizWithLinkTypes) {
     // when
     runTask("writeGraphvizLegend").build()

--- a/modular-test/src/test/kotlin/modular/mermaid/WriteMarkdownLegendTest.kt
+++ b/modular-test/src/test/kotlin/modular/mermaid/WriteMarkdownLegendTest.kt
@@ -9,10 +9,10 @@ import assertk.assertions.contains
 import assertk.assertions.exists
 import modular.test.ScenarioTest
 import modular.test.allSuccessful
+import modular.test.equalsDiffed
 import modular.test.runTask
 import modular.test.scenarios.MermaidWithLinkTypes
 import modular.test.scenarios.MermaidWithModuleTypes
-import org.junit.jupiter.api.Disabled
 import kotlin.test.Test
 
 internal class WriteMarkdownLegendTest : ScenarioTest() {
@@ -35,7 +35,6 @@ internal class WriteMarkdownLegendTest : ScenarioTest() {
     )
   }
 
-  @Disabled("https://github.com/jonapoul/modular/issues/247")
   @Test
   fun `Write markdown legend with no module types`() = runScenario(MermaidWithLinkTypes) {
     // when
@@ -45,13 +44,13 @@ internal class WriteMarkdownLegendTest : ScenarioTest() {
     assertThat(result.tasks).allSuccessful()
     val legend = resolve("modular/legend.md")
     assertThat(legend).exists()
-    assertThat(legend.readText()).contains(
+    assertThat(legend.readText()).equalsDiffed(
       """
         | Link Types | Style |
         |:--:|:--:|
         | api | Green |
         | implementation | #5555FF |
-        | compileOnly | Yellow Dotted |
+        | compileOnly | Yellow Dashed |
       """.trimIndent(),
     )
   }

--- a/modular-test/src/test/kotlin/modular/test/GradleRunners.kt
+++ b/modular-test/src/test/kotlin/modular/test/GradleRunners.kt
@@ -10,7 +10,7 @@ import java.io.File
 internal fun File.buildRunner(androidHome: File? = null): GradleRunner = GradleRunner
   .create()
   .withPluginClasspath()
-  // .withDebug(true)
+  .withDebug(false)
   .withGradleVersion(System.getProperty("test.version.gradle"))
   .withProjectDir(this)
   .apply {

